sql/plan/innerjoin.go

```diff
@@ -106,14 +106,25 @@ func (j *InnerJoin) TransformExpressionsUp(f sql.TransformExprFunc) (sql.Node, e
 	return NewInnerJoin(left, right, cond), nil
 }
 
-func (j InnerJoin) String() string {
+func (j *InnerJoin) String() string {
 	pr := sql.NewTreePrinter()
 	_ = pr.WriteNode("InnerJoin(%s)", j.Cond)
 	_ = pr.WriteChildren(j.Left.String(), j.Right.String())
 	return pr.String()
 }
 
 // Expressions implements the Expressioner interface.
-func (j InnerJoin) Expressions() []sql.Expression {
+func (j *InnerJoin) Expressions() []sql.Expression {
 	return []sql.Expression{j.Cond}
 }
+
+// TransformExpressions implements the Expressioner interface.
+func (j *InnerJoin) TransformExpressions(f sql.TransformExprFunc) error {
```

Comment:
why add this method when nodes already have `TransformExpressionsUp`?

Reply:
Because `TransformExpressionsUp` transforms expressions for this node and its childrens meanwhile this method only transform the expressions for this node.

Reply:
Ok. In any case, all nodes should be immutables, so they would need to return new instances, just as `TransformExpressionsUp` and `TransformUp` do.